### PR TITLE
 SDK - Components - Add support for the Boolean type

### LINKED ISSUE #1488: Handling of boolean input types
I have a pipeline that has a boolean parameter:

```
def test(
    param: dsl.PipelineParam = dsl.PipelineParam(name='param', value=True, param_type=TypeMeta('bool')),
) -> None:
    pass
```

When starting a new pipeline via the UI, the input field is a textbox, prefilled with the input string 'True'. Given that the input value is evaluated using `bool(value)`, the only way to start the pipeline with a `False` value is to leave the input textbox empty.

It seems that currently the pipeline does not gracefully handle boolean parameters?

### PR DESCRIPTION
**Pull request by Ark-kun**
This PR is based on https://github.com/kubeflow/pipelines/pull/1934 Merging that PR will make this PR very small.
Fixes #1488

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1936)

### CODE REVIEW

File: sdk/python/kfp/components/_data_passing.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        return serializer(value)
+        #try:
+        #    return serializer(value)
+        #except Exception as e:
+        #    warnings.warn('Failed to serialize the value "{}" of type "{}" to type "{}". The value will be serialized using str(). Exception: {}'.format(
+        #        str(value),
+        #        str(type(value).__name__),
+        #        str(type_name),
+        #        str(e),
+        #    ))
```
Comment (ajchili) [inline sdk/python/kfp/components/_data_passing.py]: Should this be kept in the code base?

Comment (Ark-kun) [reply sdk/python/kfp/components/_data_passing.py]: I'm not yet fully sure what's the best way to handle serialization failures.

Comment (ajchili) [reply sdk/python/kfp/components/_data_passing.py]: Keeping it then makes sense.

Comment (Ark-kun) [reply sdk/python/kfp/components/_data_passing.py]: I've fixed this place. (BTW, it's part of the https://github.com/kubeflow/pipelines/pull/1934 PR)

File: sdk/python/kfp/components/_python_op.py [hunk 2]
```diff
@@ -146,10 +147,17 @@ def annotation_to_type_struct(annotation):
         if not annotation or annotation == inspect.Parameter.empty:
```
Comment (ajchili) [inline sdk/python/kfp/components/_python_op.py]: I am not sure if this is within the scope of this PR but should there be type definitions for parameters?

File: sdk/python/kfp/dsl/types.py [hunk 3]
```diff
@@ -94,6 +98,35 @@ class InconsistentTypeException(Exception):
 	'''InconsistencyTypeException is raised when two types are not consistent'''
 	pass
 
+
+class InconsistentTypeWarning(Warning):
+	'''InconsistentTypeWarning is issued when two types are not consistent'''
+	pass
+
+
+def verify_type_compatibility(given_type, expected_type, error_message_prefix : str = ''):
```
Comment (ajchili) [inline sdk/python/kfp/dsl/types.py]: Should all parameters have a defined type?

File: sdk/python/kfp/components/_python_op.py (hunk 2 above)
Comment (Ark-kun) [reply sdk/python/kfp/components/_python_op.py]: In this case, the python annotation can be any type.

File: sdk/python/kfp/dsl/types.py (hunk 3 above)
Comment (Ark-kun) [reply sdk/python/kfp/dsl/types.py]: Fixed in https://github.com/kubeflow/pipelines/pull/1934

**Comment by Ark-kun** [comment]
> Small syntax suggestions, otherwise LGTM

Can you please LGTM https://github.com/kubeflow/pipelines/pull/1934 (the refactoring part of this PR)?

**Comment by Ark-kun** [comment]
> Small syntax suggestions, otherwise LGTM

Now this small PR is ready.

**Pull request merged**